### PR TITLE
LRU caching implementation for sessions and last resort prekeys

### PR DESCRIPTION
**Pull request by Yserz**

### CODE REVIEW

File: src/main/ts/Cryptobox.ts [hunk 1]
```diff
@@ -47,13 +75,25 @@ export class Cryptobox {
...
           this.logger.log(`Loading last resort PreKey...`);
-          return this.store.load_prekey(Proteus.keys.PreKey.MAX_PREKEY_ID);
+          return Promise.resolve().then(() => {
+            return this.load_prekey_from_cache(Proteus.keys.PreKey.MAX_PREKEY_ID);
+          }).then((prekey) => {
+            if (prekey === undefined) {
+              return this.store.load_prekey(Proteus.keys.PreKey.MAX_PREKEY_ID)
+            } else {
+              return prekey;
+            }
```
Comment (herrmannplatz) [inline src/main/ts/Cryptobox.ts]: return prekey || this.store.load_prekey(Proteus.keys.PreKey.MAX_PREKEY_ID)

File: src/main/ts/Cryptobox.ts [hunk 2]
```diff
@@ -47,13 +75,25 @@ export class Cryptobox {
           this.identity = identity;
           this.logger.log(`Initialized Cryptobox with an identity which has the following fingerprint "${this.identity.public_key.fingerprint()}".`, this.identity);
           this.logger.log(`Loading last resort PreKey...`);
-          return this.store.load_prekey(Proteus.keys.PreKey.MAX_PREKEY_ID);
+          return Promise.resolve().then(() => {
+            return this.load_prekey_from_cache(Proteus.keys.PreKey.MAX_PREKEY_ID);
+          }).then((prekey) => {
```
Comment (bennyn) [inline src/main/ts/Cryptobox.ts]: `.then((prekey)` has no type.

File: src/main/ts/Cryptobox.ts [hunk 3]
```diff
@@ -1,19 +1,18 @@
...
-  // TODO: Limit the amount of items in cache
   public EVENT = {
     NEW_PREKEYS: "new-prekeys"
   };
 
-  private cachedPreKeys: LRUCache = new LRUCache(1000);
-  private cachedSessions: LRUCache = new LRUCache(1000);
+  private cachedPreKeys: LRUCache;
+  private cachedSessions: LRUCache;
   private channel = postal.channel("cryptobox");
```
Comment (herrmannplatz) [inline src/main/ts/Cryptobox.ts]: 👌

File: src/main/ts/Cryptobox.ts [hunk 4]
```diff
@@ -1,19 +1,18 @@
...
 export class Cryptobox {
-  // TODO: Limit the amount of items in cache
   public EVENT = {
     NEW_PREKEYS: "new-prekeys"
   };
 
-  private cachedPreKeys: LRUCache = new LRUCache(1000);
-  private cachedSessions: LRUCache = new LRUCache(1000);
+  private cachedPreKeys: LRUCache;
+  private cachedSessions: LRUCache;
```
Comment (herrmannplatz) [inline src/main/ts/Cryptobox.ts]: 👌

**Pull request merged**